rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -320,6 +323,17 @@ rmw_destroy_node(rmw_node_t * node)
     allocator->deallocate(node_data, allocator->state);
   }
 
+  // Erase the node from the set of session_nodes and close the Zenoh
+  // session if this is the last node.
+  node->context->impl->session_nodes_.erase(node);
+  if (node->context->impl->session_nodes_.empty()) {
```

Comment:
Shouldn't you also test if `node->context->impl->is_shutdown == true` ?
I mean could it happen that some process close all its Nodes without calling `rcl_shutdown()`, but re-creating Nodes later ?